lms/djangoapps/courseware/views/views.py

```diff
@@ -1525,7 +1525,7 @@ def financial_assistance_form(request):
             Q(_expiration_datetime__isnull=True) | Q(_expiration_datetime__gt=datetime.now(UTC())),
             course_id=enrollment.course_id,
             mode_slug=CourseMode.VERIFIED
-        ).exists()
+        ).exists() and CourseOverview.objects.filter(id=enrollment.course_id, eligible_for_financial_aid=True).exists()
```

Comment:
This is very inefficient. You will query `CourseOverview` for every course run in which the user is enrolled. You should first get the valid enrollments. Use those course run keys to query `CourseOverview` once and do the final filtering in Python.